src/test/java/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStepTest.java

```diff
@@ -87,12 +86,9 @@ public void abort() throws Exception {
         WorkflowRun b = foo.scheduleBuild2(0).getStartCondition().get();
 
         // at this point the file should be being touched
-        waitForCond(15000, tmp, new Predicate<File>() {
-            @Override
-            public boolean apply(File tmp) {
-                return tmp.exists();
-            }
-        });
+        while (!tmp.exists()) {
+            Thread.sleep(100);
+        }
 
         b.getExecutor().interrupt();
```

Comment:
Note that this can flake out below, probably because the interruption does not take effect synchronously, so the file might still be modified for a few moments afterward.